tensorboard/data/experimental/experiment_from_dev.py

```diff
@@ -0,0 +1,162 @@
+from tensorboard.uploader.proto import server_info_pb2
+from tensorboard.util import grpc_util
+
+
+DEFAULT_ORIGIN = "https://tensorboard.dev"
+
+
+def import_pandas():
+    """Import panads, guarded by a user-friendly error message on failure."""
+    try:
+        import pandas
+    except ImportError:
+        raise ImportError(
+            "The get_scalars() feature requires the pandas package, "
+            "which does not seem to be available in your Python "
+            "environment. You can install it with command: \n\n"
+            "  pip install pandas"
+        )
+    return pandas
+
+
+class ExperimentFromDev(base_experiment.BaseExperiment):
+    """Implementation of BaseExperiment, specialized for tensorboard.dev."""
+
+    def __init__(self, experiment_id, api_endpoint=None):
+        super(ExperimentFromDev, self).__init__()
+        self._experiment_id = experiment_id
+        self._api_client = get_api_client(api_endpoint=api_endpoint)
+
+    def get_scalars(self, runs_filter=None, tags_filter=None, pivot=True):
+        if runs_filter is not None:
+            raise NotImplementedError(
+                "runs_filter support for get_scalars() is not implemented yet."
+            )
+        if tags_filter is not None:
+            raise NotImplementedError(
+                "tags_filter support for get_scalars() is not implemented yet."
+            )
+        pandas = import_pandas()
+
+        request = export_service_pb2.StreamExperimentDataRequest()
+        request.experiment_id = self._experiment_id
+        read_time = time.time()
+        util.set_timestamp(request.read_timestamp, read_time)
+        # TODO(cais, wchargin): Use another rpc to check for staleness and avoid
+        # a new StreamExperimentData rpc request if data is not stale.
+        stream = self._api_client.StreamExperimentData(
+            request, metadata=grpc_util.version_metadata()
+        )
+
+        runs = []
+        tags = []
+        steps = []
+        wall_times = []
+        values = []
+        for response in stream:
+            # TODO(cais, wchargin): Display progress bar.
+            metadata = base64.b64encode(
+                response.tag_metadata.SerializeToString()
+            ).decode("ascii")
+            num_values = len(response.points.values)
+            runs.extend([response.run_name] * num_values)
+            tags.extend([response.tag_name] * num_values)
+            steps.extend(list(response.points.steps))
+            wall_times.extend(
+                [t.ToNanoseconds() / 1e9 for t in response.points.wall_times]
+            )
+            values.extend(list(response.points.values))
+
+        dataframe = pandas.DataFrame(
+            {
+                "run": runs,
+                "tag": tags,
+                "step": steps,
+                "wall_time": wall_times,
+                "value": values,
+            }
+        )
+        if pivot:
```

Comment:
nit: Can we just `if not pivot: return dataframe` to avoid keeping two
codepaths in mind? (Also avoids indenting the whole block.)